b2share/modules/schemas/api.py

```diff
@@ -194,7 +196,53 @@ def create_block_schema(cls, community_id, name, id_=None):
             raise InvalidBlockSchemaError() from e
         return block_schema
 
-    def create_version(self, json_schema):
+    def update(self, data, clear_fields=False):
+        """Update multiple fields of the schema at the same time.
+        If the update fails, none of the schema's fields are modified.
+        Args:
+            data (dict): it replaces the given values.
+            clear_fields (bool): if True, set not specified fields to None.
+        Returns:
+            :class:`BlockSchema`: self
+        Raises:
+            b2share.modules.schemas.InvalidBlockSchemaError: The
+                schema update failed because the resulting schema is
+                not valid.
+        """
+        try:
+            with db.session.begin_nested():
+                if clear_fields:
+                    for field in ['name']:
+                        setattr(self.model, field, data.get(field, None))
+                else:
+                    for key, value in data.items():
+                        setattr(self.model, key, value)
+                db.session.merge(self.model)
+        except sqlalchemy.exc.IntegrityError as e:
+            raise InvalidBlockSchemaError() from e
+        return self
+
+    def patch(self, patch):
+        """Update the schema's metadata with a json-patch.
+        Args:
+            patch (str): json-patch which can modify the following fields:
+                name, description, logo.
+        Returns:
+            :class:`BlockSchema`: self
+        Raises:
+            jsonpatch.JsonPatchConflict: the json patch conflicts on the
+                block schema.
+            jsonpatch.InvalidJsonPatch: the json patch is invalid.
+            b2share.modules.block_schemas.errors.InvalidBlockSchemaError: The
+                block schema patch failed because the resulting block schema is
+                not valid.
+        """
+        data = apply_patch({'name': self
+        .model.name}, patch, True)
```

Comment:
Could you please split the line after `.model.name}`. Splitting it on the dot makes the code less readable. It would also be better if the second line was indented properly.